gulpfile.js

```diff
@@ -218,33 +239,37 @@ gulp.task('build-frontend-games-html', ['clean-frontend'], function () {
 
 });
 
-gulp.task('build-frontend-js-vendor', ['clean-frontend'], function () {
-  
-  var b = browserify({
-    entries: buildConfig.frontend.js.vendor.src,
-    debug: true
-  });
+function buildBrowserifyBundle(bundleOptions) {
 
-  return b.bundle()
-    .pipe(source('vendor.js'))
-    .pipe(buffer())
-    .pipe(sourcemaps.init({loadMaps: true}))
-//      .pipe(uglify())
-//      .on('error', util.log)
-    .pipe(sourcemaps.write('./'))
-    .pipe(gulp.dest(buildConfig.frontend.js.vendor.dest));
+    var b = browserify({
+      entries: bundleOptions.entries,
+      debug: true
+    });
 
-});
+    b.require(bundleOptions.require);
+
+    return b.bundle()
+      .pipe(source(bundleOptions.output))
+      .pipe(buffer())
+      .pipe(sourcemaps.init({loadMaps: true}))
+  //      .pipe(uglify())
+  //      .on('error', util.log)
+      .pipe(sourcemaps.write('./'))
+      .pipe(gulp.dest(bundleOptions.dest));
+}
 
-gulp.task('build-frontend-js-project', ['test-frontend', 'clean-frontend'], function () {
+gulp.task('build-frontend-js-bundles', ['test-frontend', 'clean-frontend'], function () {
   
-  return gulp.src(buildConfig.frontend.js.project.src)
-    .pipe(gulp.dest(buildConfig.frontend.js.project.dest));
+  var tasks = _.map(buildConfig.frontend.js.bundles, function (bundleOptions) {
+    return buildBrowserifyBundle(bundleOptions);
+  });
 
+  // create a merged stream
+  return es.merge.apply(null, tasks);
```

Comment:
@greybubble This merges the various bundle build streams into one, which we return.  This is sort of like combining promises.

We return the combined stream, which will end when all of the individual streams end.  Since `Gulp` won't move on to the next task until this stream completes, we can be sure that we will have proper task ordering.